### ISSUE TITLE
deprecate in favor of a plugin

### ISSUE THREAD

**Issue by brianium** [peridot-php/peridot-code-coverage-reporters#8]
This library is not as useful as a collection of reporters. It really should just function as a plugin that honors the reporter being used. That way we get output + coverage :)

I imagine most of the code would be the same, but we would use the same events, but stop using the reporter api to hijack the output. 

I would vote creating a new package and just deprecating this one adding a "recommended" package of the new solution to the composer.json

**Comment by shadowhand** [comment]
Yes please. The current state of code coverage with peridot is this:

- `phpdbg` is not available as a homebrew package (at least not in core + php sources)
- this package requires the usage of `phpdbg`
- this package doesn't actually generate any coverage itself

As it stands right now, this package looks like a red herring because it doesn't *just work*. What I want is:

- a plugin I can install and enable to get coverage, ala phpunit

**Comment by ezzatron** [comment]
@shadowhand I hear your frustration, and I know this package, and particularly the docs, are not perfect. But to be fair:

- This package doesn't require `phpdbg`, it simply uses [sebastianbergmann/php-code-coverage], which can use *either* `phpdbg` or `xdebug`.
- It doesn't generate coverage, but it does handle the required glue to add information such as which tests cover which lines to the HTML coverage report generated by [sebastianbergmann/php-code-coverage].
- `phpdbg` can be installed via brew using the `--with-phpdbg` flag when installing PHP. See [my dotfiles](https://github.com/ezzatron/dotfiles/blob/master/setup/2-php.bash#L51) for an example. It's slow because it requires compilation, but it works fine.
- There's not a lot of difference here to how PHPUnit coverage works. As far as I know, you need either `phpdbg` or `xdebug` to generate coverage there too. It is the same coverage library after all.

I think this particular GitHub issue was more about the fact that you couldn't (at the time), run coverage and see dot/spec output at the same time. This has since been addressed by adding support for multiple reporters.

[sebastianbergmann/php-code-coverage]: https://github.com/sebastianbergmann/php-code-coverage
